apps/dapp/pages/dao/[address]/index.tsx

```diff
@@ -104,12 +132,31 @@ const InnerDAOHome = () => {
   const { t } = useTranslation()
   const router = useRouter()
 
-  const { coreAddress, name } = useDaoInfoContext()
+  const { coreAddress, name, proposalModules } = useDaoInfoContext()
   const {
     components: { Membership },
   } = useVotingModuleAdapter()
   const { isMember } = useVotingModule(coreAddress, { fetchMembership: true })
 
+  const useDepositInfoHooks = useMemo(
+    () =>
+      proposalModules.map(
+        (proposalModule) =>
+          matchAndLoadCommon(proposalModule, {
+            coreAddress,
+            Loader,
+            Logo,
+          }).hooks.useDepositInfo
+      ),
+    [coreAddress, proposalModules]
+  )
+  const proposalModuleDepositInfos = useDepositInfoHooks
+    .map((useDepositInfo) =>
+      // eslint-disable-next-line react-hooks/rules-of-hooks
+      useDepositInfo?.()
+    )
+    .filter(Boolean) as CheckedDepositInfo[]
```

Comment:
What's going on in this line? This has "magical chained method" energy. Perhaps a comment would help?